filebeat/crawler/prospector.go

```diff
@@ -35,9 +36,42 @@ func (p *Prospector) Init() error {
 		return err
 	}
 
+	var prospectorer Prospectorer
+
+	switch p.ProspectorConfig.Harvester.InputType {
+	case cfg.StdinInputType:
+		prospectorer = ProspectorStdin{
+			Prospector: p,
```

Comment:
Registrar is currently also used in stdin to send the event. 

So what you suggest is that we have a function "OnEvent" in Prospectorer which is triggered by Prospector. So we get rid of the for loops inside the Prospectorer. This would mean we would also have to introduce an Init / Setup function for Prospectorer, as there are things that should only be run once.

I like the idea.